bot/cogs/candy_collection.py

```diff
@@ -0,0 +1,210 @@
+        """Get the message from it's ID."""
+
+        try:
+            o = discord.Object(id=msg_id + 1)
+            # Use history rather than get_message due to
+            #         poor ratelimit (50/1s vs 1/1s)
+            msg = await self.hacktober_channel.history(limit=1, before=o).next()
+
+            if msg.id != msg_id:
+                return None
+
+            return msg
+
+        except Exception:
+            return None
+
+    async def hacktober_channel(self):
+        """Get #hacktoberbot channel from it's id"""
+        return self.bot.get_channel(id=self.HACKTOBER_CHANNEL_ID)
+
+    async def remove_reactions(self, reaction):
+        """Remove all candy/skull reactions"""
+
+        try:
+            async for user in reaction.users():
+                await reaction.message.remove_reaction(reaction.emoji, user)
+
+        except discord.HTTPException:
+            pass
+
+    async def send_spook_msg(self, author, channel, candies):
+        """Send a spooky message"""
+        e = discord.Embed(colour=author.colour)
+        e.set_author(name="Ghosts and Ghouls and Jack o' lanterns at night; "
+                          f"I took {candies} candies and quickly took flight.")
+        await channel.send(embed=e)
+
+    def save_to_json(self):
+        """Save json to the file."""
+        with open(json_location, 'w') as outfile:
+            json.dump(self.candy_json, outfile)
+
+    @commands.command()
+    async def candy(self, ctx):
+        """Get the candy leaderboard and save to json when this is called"""
+
+        # use run_in_executor to prevent blocking
+        thing = functools.partial(self.save_to_json)
+        save = await self.bot.loop.run_in_executor(None, thing)
+
+        emoji = (
+            '\N{FIRST PLACE MEDAL}',
+            '\N{SECOND PLACE MEDAL}',
+            '\N{THIRD PLACE MEDAL}',
+            '\N{SPORTS MEDAL}',
+            '\N{SPORTS MEDAL}'
+        )
+
+        top_sorted = sorted(self.candy_json['records'], key=lambda k: k.get('record', 0), reverse=True)
+        top_five = top_sorted[:5]
+
+        usersid = []
+        records = []
+        for record in top_five:
+            usersid.append(record['userid'])
+            records.append(record['record'])
+
+        value = '\n'.join(f'{emoji[index]} <@{usersid[index]}>: {records[index]}'
+                          for index in range(0, len(usersid))) or 'No Candies'
+
+        e = discord.Embed(colour=discord.Colour.blurple())
+        e.add_field(name="Top Candy Records", value=value, inline=False)
+        e.add_field(name='\u200b',
+                    value=f"Candies will randomly appear on messages sent. "
+                          f"\nHit the candy when it appears as fast as possible to get the candy! "
+                          f"\nBut beware the ghosts...",
+                    inline=False)
+        await ctx.send(embed=e)
+
+
+def setup(bot):
+    bot.add_cog(CandyCollection(bot))
+
```

Comment:
From the linter:
`bot/cogs/candy_collection.py:210:1: W391 blank line at end of file`

Reply:
wait I need it or not?

Reply:
pep8 pulls me up if I dont have it lol